airflow/providers/amazon/aws/hooks/redshift_cluster.py

```diff
@@ -157,16 +158,24 @@ def create_cluster_snapshot(
         )
         return response["Snapshot"] if response["Snapshot"] else None
 
-    def get_cluster_snapshot_status(self, snapshot_identifier: str, cluster_identifier: str):
+    def get_cluster_snapshot_status(self, snapshot_identifier: str, cluster_identifier: str | None = None):
         """
         Return Redshift cluster snapshot status. If cluster snapshot not found return ``None``
 
         :param snapshot_identifier: A unique identifier for the snapshot that you are requesting
-        :param cluster_identifier: The unique identifier of the cluster the snapshot was created from
+        :param cluster_identifier: (deprecated) The unique identifier of the cluster
+            the snapshot was created from
```

Comment:
Can you please share some context?
The PR description says the motivation is issues with system tests but there should be more to it?
Why are we deprecating this parameter? If boto3 accept this as valid input why should we prevent users from using it?

Reply:
The change was made as a result of debugging during testing for the redshift system test. It was not originally in scope of the refactoring of the `RedshiftResumeClusterOperator` and `RedshiftPauseClusterOperator`. I included it in this PR because the `get_cluster_snapshot_status` function does not work as written.

>If boto3 accept this as valid input why should we prevent users from using it?

Boto3 does not accept both inputs at the same time. If both are provided, we get an `InvalidParameterCombination` exception.

Reply:
In that case, maybe leave the parameter in and add a block at the beginning of the method along the lines of 

```
if not(cluster_identifier ^ snapshot_identifier):
    raise SomeException("cluster_identifier or snapshot_identifier must be included, but not both")
```    

The ^ is an XOR operator, which returns true if one and only one of those exists.
![image](https://user-images.githubusercontent.com/1920178/200046033-1acf98d9-fc12-4562-8000-6d8286bc604b.png)